### PR TITLE
[dependency] update to GraphQL Java 17

### PR DESCRIPTION
**Pull request by dariuszkuc**
### :pencil: Description

Update to `graphql-java` 17. Subsequent PRs will address the changes in the API (e.g. new GraphQL context).

### CODE REVIEW

File: generator/graphql-kotlin-federation/src/test/kotlin/com/expediagroup/graphql/generator/federation/execution/ServiceQueryResolverTest.kt [hunk 1]
```diff
@@ -105,14 +105,14 @@ class ServiceQueryResolverTest {
             .coercing(CustomScalarCoercing()).build()
 
         private class CustomScalarCoercing : Coercing<CustomScalar, String> {
-            override fun parseValue(input: Any?): CustomScalar = CustomScalar(serialize(input))
+            override fun parseValue(input: Any): CustomScalar = CustomScalar(serialize(input))
 
-            override fun parseLiteral(input: Any?): CustomScalar {
+            override fun parseLiteral(input: Any): CustomScalar {
                 val customValue = (input as? StringValue)?.value ?: throw CoercingParseValueException("CustomScalar value is null")
```
Comment (smyrick) [inline generator/graphql-kotlin-federation/src/test/kotlin/com/expediagroup/graphql/generator/federation/execution/ServiceQueryResolverTest.kt:111]: Is this exception still needed or should we change the message. Also this is a test so I suppose we could delete

**Comment by smyrick** [comment]
So are we just going to continue to use the deprecated context method for now and migrate and delete our custom interface in v6? Should we deprecate that interface then?

File: generator/graphql-kotlin-federation/src/test/kotlin/com/expediagroup/graphql/generator/federation/execution/ServiceQueryResolverTest.kt (hunk 1 above)
Comment (dariuszkuc) [reply generator/graphql-kotlin-federation/src/test/kotlin/com/expediagroup/graphql/generator/federation/execution/ServiceQueryResolverTest.kt:111]: per Javadocs when unable to resolve a scalar then it should throw the `CoercingParseValueException`

**Comment by dariuszkuc** [comment]
@smyrick I was thinking that we might wait for the `spring-graphql` GA release to determine what to do with the context.

**Review by smyrick — APPROVED** [review]

**Pull request merged**